### DISCUSSION TITLE
Ptah - Leveraging Substrate Rust nodes and replace it by the Rust Cairo VM

### DISCUSSION THREAD

**Discussion by louisguthmannStarkWare** [stark-rocket/.github#1] [Ideas]
StarkNet is slowly turning in a pretty Rust centric environment. In the future, StarkNet will decentralise via its own set of nodes.

An idea to speed up this network creation is to reuse the Substrate Rust stack and swap the WASM runtime by the Rust Cairo VM with StarkNet OS implemented.

![image](https://user-images.githubusercontent.com/51161328/205751993-3f1196f2-b23f-45fd-8f6a-3b7bb4961694.png)

This way we will be able to quickly spin up simulations nodes, side chains and eventually, use it as nodes for StarkNet

**Comment by AbdelStark** [comment]
Being worked on with https://github.com/keep-starknet-strange/kaioshin
